### PR TITLE
mkcloud: libvirt nodesetup restructuring and cleanup

### PR DESCRIPTION
**Pull request by jdsn**
The new function to create libvirt configs is now being used for regular nodes as well as lonely nodes.
Background is 
- support DRBD volumes on nodes 1 and 2 even if they are lonely nodes - this is required for an HA setup consisting of lonely nodes only
- remove code duplication
- easier introduction of new node types (will follow soon) without having to duplicate code

### CODE REVIEW

File: scripts/lib/mkcloud-libvirt.sh [hunk 1]
```diff
@@ -334,26 +334,49 @@ function libvirt_do_onhost_deploy_image()
...
+                drbdnode_mac_vol="${drbdnode_mac_vol}+${mac}#${drbd_serial}"
+                drbdnode_mac_vol="${drbdnode_mac_vol#+}"
+            fi
+        fi
+
+        local bootorder=1
+        if [[ " $(nodes ids lonely) " =~ " $number " ]] ; then
+            bootorder=3
+        fi
+        ${mkcloud_lib_dir}/libvirt/compute-config "$cloud" "$number"\
```
Comment (bmwiedemann) [inline scripts/lib/mkcloud-libvirt.sh]: testrun failed from empty mkcloud_lib_dir. was scripts_lib_dir before.
```
14:57:44 /root/workspace/openstack-mkcloud@2/SUSE-Cloud.automation.RnT3H3/automation/scripts/lib/mkcloud-libvirt.sh: line 360: /libvirt/compute-config: No such file or directory
14:57:44 + [[ normal = lonely ]]
14:57:44 + /libvirt/vm-start /tmp/vd2-node2.xml
14:57:44 /root/workspace/openstack-mkcloud@2/SUSE-Cloud.automation.RnT3H3/automation/scripts/lib/mkcloud-libvirt.sh: line 379: /libvirt/vm-start: No such file or directory
```

Comment (jdsn) [inline scripts/lib/mkcloud-libvirt.sh]: thanks, these patches originate from times before we renamed the variable

**Comment by jdsn** [comment]
rebased on latest master (without tests yet)

**Comment by jdsn** [comment]
added @rhafer @abel-navarro to have a look at the changes from the ironic perspective

File: scripts/lib/mkcloud-libvirt.sh [hunk 2]
```diff
@@ -381,72 +381,82 @@ function libvirt_do_onhost_deploy_image()
...
-        local ironic_disk
-        ironic_disk="$vdisk_dir/${cloud}.node$i"
-
-        $sudo lvdisplay "$ironic_disk" || \
-            _lvcreate "${cloud}.node$i" "${ironicnode_hdd_size}" "$cloudvg"
-
-        libvirt_vm_start /tmp/${ironic_node}.xml
+    for i in $@ ; do
+        case $nodetype in
+            local root_disk
```
Comment (bmwiedemann) [inline scripts/lib/mkcloud-libvirt.sh]: mkcloud failed, because you cannot use 'local' within a case.

Comment (jdsn) [inline scripts/lib/mkcloud-libvirt.sh]: fixed

**Comment by jdsn** [comment]
triggered a bunch of additional gatings to cover all the different deployments that we have and that are changed by this PR

**Comment by toabctl** [comment]
over 2 years old. closing.

**Pull request closed without merging**